src/convert.jl

```diff
@@ -225,14 +225,29 @@ end
 function sexp2julia(rl::RList)
     if isdataframe(rl)
         # FIXME add force_missing option to control whether always convert to Union{T, Missing}
-        DataFrame(Any[isa(col, RAltRep) ? sexp2julia(col) : jlvec(col, false) for col in rl.data],
-                  identifier.(names(rl)), makeunique=true)
+        cols = Any[isa(col, RAltRep) ? sexp2julia(col) : jlvec(col, false) for col in rl.data]
+        nms = identifier.(names(rl))
+        obj = DataFrame(cols, nms, makeunique=true)
+        meta = DataAPI.metadata(obj)
+        sizehint!(meta, length(rl.attr))
+        for (key, val) in pairs(rl.attr)
+            key in ("names", "class", "row.names") && continue
+            meta[key] = sexp2julia(val)
+        end
+        for (col, name) in zip(rl.data, nms)
+            colmeta = DataAPI.metadata(obj, name)
```

Comment:
```suggestion
            colmeta = DataAPI.colmetadata(obj, name)
```

You don't check for "class"/"name" attributes here. Is it intentional? (I have no strong preference what to do in such cases. I don't know if any R data frame extension makes special use of them. But they could be present as a side product of "hand-crafted" data frame creation).